opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -68,6 +69,18 @@ def wrapper(  # type: ignore[misc]
     return typing.cast(_F, wrapper)  # type: ignore[misc]
 
 
+def _create_key(keyname: str) -> str:
```

Comment:
This is just a suggestion if its easy enough to implement – the return type for the key seems like a great fit for a [NewType](https://docs.python.org/3/library/typing.html#newtype) (e.g. `ContextKey = NewType('ContextKey', str)`). Then we can annotate where a `ContextKey` is expected so its clear to users and mypy can enforce it.

Then you can add an overload for `set_value(key: ContextKey, ...)` and we can remove the original str key overload in the next major version.